scipy/signal/tests/test_fir_filter_design.py

```diff
@@ -652,3 +652,103 @@ def test_hilbert(self):
              -0.014977068692269, -0.158416139047557]
         m = minimum_phase(h, 'hilbert', n_fft=2**19)
         xp_assert_close(m, k, rtol=2e-3)
+
+
+class TestFwind1:
+    def test_invalid_args(self):
+        with pytest.raises(ValueError, 
+                           match="hsize must be a 2-element tuple or list"):
+            fwind1((50,), window=(("kaiser", 5.0), "boxcar"), fc=0.4)
+        
+        with pytest.raises(ValueError, 
+                           match="window must be a 2-element tuple or list"):
+            fwind1((51, 51), window=("hamming",), fc=0.5)
+        
+        with pytest.raises(ValueError, 
+                           match="window must be a 2-element tuple or list"):
+            fwind1((51, 51), window="invalid_window", fc=0.5)
+
+    def test_filter_design(self):
+        hsize = (51, 51)
+        window = (("kaiser", 8.0), ("kaiser", 8.0))
+        fc = 0.4
+        taps_kaiser = fwind1(hsize, window, fc)
+        assert taps_kaiser.shape == (51, 51)
+
+        window = ("hamming", "hamming")
+        taps_hamming = fwind1(hsize, window, fc)
+        assert taps_hamming.shape == (51, 51)
+
+    def test_impulse_response(self):
+        hsize = (31, 31)
+        window = ("hamming", "hamming")
+        fc = 0.4
+        taps = fwind1(hsize, window, fc)
+
+        impulse = np.zeros((63, 63))
+        impulse[31, 31] = 1
+
+        response = convolve2d(impulse, taps, mode='same')
+
+        expected_response = taps
+        xp_assert_close(response[16:47, 16:47], expected_response, rtol=1e-5)
+
+    def test_frequency_response(self):
+        hsize = (31, 31)
+        window = ("hamming", "hamming")
+        fc = 0.4
+        taps = fwind1(hsize, window, fc)
+
+        freq_response = fftshift(fft2(taps))
+
+        magnitude = np.abs(freq_response)
+        assert np.isclose(magnitude.max(), 1.0, atol=0.01), (
```

Comment:
I think this should be an `xp_assert_close` as this will give a better error message if failing